llvm/lib/Transforms/Utils/ControlFlowUtils.cpp

```diff
@@ -342,3 +345,56 @@ std::pair<BasicBlock *, bool> ControlFlowHub::finalize(
 
   return {FirstGuardBlock, true};
 }
+
+template <typename TI, typename T>
+static bool updateCycleLoopInfo(TI *LCI, BasicBlock *CallBrBlock,
```

Comment:
Could you add a doxygen deswcription about what this is updating?

Reply:
done